src/libbson/src/bson/bson.c

```diff
@@ -3174,7 +3238,9 @@ _bson_as_json_visit_all (const bson_t *bson,
       return NULL;
    }
 
-   bson_string_append (state.str, " }");
+   if (!state.max_len_reached) {
```

Comment:
Should this check that there is room for the last two characters? E.g. `state.str->len + 2 <= state.max_len`?